### ISSUE TITLE
icd_history: Explain automatic "back" and "forward"

### ISSUE THREAD

**Issue by edonnachie** [edonnachie/ICD10gm#8]
The differentiation between automatic and non-automatic transitions is a potential source of .confusion and error. Improve documentation of icd_history function and vignette to explain:

- icd_history, by default, only follows automatic transitions
- automatic transitions mean there is a unique mapping from one code to another. If a new version introduces new codes to describe specific subtypes of a disease (e.g. a new fifth digit), the new code can always be mapped back to the old, less specific code (i.e. the corresponding four-digit code). A substantial change in the underlying medical classification necessarily introduces a break in the classification that cannot be solved on the basis of metadata alone. For users of secondary data, this means that the implications and possible solutions must be considered carefully in order to in the context of the study design. Custom transitions may be helpful in such cases.

Examples:

- __automatic forward and backward__: In 2018 Causalgia was coded under G56.4 (Causalgia). In 2019, there is an automatic transition to G90.6 (Complex regional pain syndrome (CRPS) of the extremity, Type II), indicating direct equivalence of the two codes.

- __automatic backward__: In 2019, the fourth digit of the code K58 (irritable bowel syndrome, IBS) was changed to reflect current guidelines. Whereas previously two codes were available (K58.0 "with diarrhoea" and K58.9 "without diarrhoea"), the new classification uses four codes (K58.1 "diarrhoea predominant", K58.2 "constipation predominant", K58.3 "mixed" and K58.8 "other/unspecified"). As a result, there is no natural transition between the two ICD versions. However, a "diarrhoea predominant" IBS implies "IBS with diarrhoea" such that an automatic backward transition from K58.1 to K58.0 is possible. Similarly, a "constipation predominant" IBS (2019 coding: K58.2) is assumed to imply "IBS without diarrhoea", resulting in a second automatic transition, albeit with questionable logic.

- __automatic forward__: In 2018, the code S32.83 (Fracture of the pelvis, part not further specified) existed alongside the code S32.89 (Fracture of the pelvis, other or multiple parts). In 2019, the first code is removed and transitioned automatically to the second (which now reads "other and unspecified parts"). An automatic backward transition is not appropriate because only a subset of the S32.89 codes were previously coded by S32.83 (i.e. a fracture of multiple parts should not be transitioned backwards).

**Comment by ghost** [comment]
So if I understand this correctly, when converting older ICD codes from diagnosis histories to a uniform standard (lets say ICD10gm 2021) one can only use the transitions with automatic forward (compatibility) to retain (construct) valditiy of the codes?

**Comment by edonnachie** [comment]
Yes. As I understand it, an automatic transition means that the same diagnosis is now coded by a different key. If there is no ambiguity in the transition, it can be done automatically.

Sometimes, there might be several new codes to replace the old one, which means that an automatic transition is not possible. In such cases, we might want to add a custom transition to define how this should be handled.
